resource/graph_node_test.go

```diff
@@ -183,6 +239,9 @@ func lifecycleTest(t *testing.T, node *resource.GraphNode, initialDeps []string)
 	_, err = node.Resource()
 	test.That(t, err, test.ShouldNotBeNil)
 	test.That(t, err.Error(), test.ShouldContainSubstring, "not initialized")
+	// TODO(RSDK-7928): we might want to make this transition a node to an "unconfigured"
+	// or "removing" state.
+	rs = verifySameState(t, node, resource.NodeStateReady, rs)
```

Comment:
making `node.Close` transition to one of the states in the comments results in changes in reconfiguration results, so I'd like to punt the decision to do this for another PR and keep this one as a strict no-op